### PR TITLE
Add schema validation

### PR DESCRIPTION
**Pull request by alexhung**
Per security pen testing:
- Add Xray event payload schema validation
- Add back security issue test files

### CODE REVIEW

File: eventProcessor/app.js [hunk 1]
```diff
@@ -105,10 +109,11 @@ export async function lambdaHandler(event) {
       isBase64Encoded: false,
     };
   } catch (err) {
+    err.statusCode ??= 500;
     logger.error('failed to process Xray event', { err });
     response = {
       statusCode: err.statusCode,
-      body: err,
+      body: err.toString(),
```
Comment (danielmkn) [inline eventProcessor/app.js:116]: why `.toString()`?

Comment (alexhung) [reply eventProcessor/app.js:116]: We don't necessary want the entire `err` object to be part of the response. `toString()` ensures only the textual part is used. This also allows any custom `Error` type to customize their own `toString()` output.

**Review by mritunjaykumar — APPROVED** [review]
LGTM

**Review by MahithaB — APPROVED** [review]

**Review by danielmkn — APPROVED** [review]

**Review by shrajfr12 — APPROVED** [review]
lgtm

**Pull request merged**